up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto

```diff
@@ -41,331 +41,300 @@ service uDiscovery {
-  // you will get PERMISSION_DENIED and no nodes will be deleted.
-  rpc DeleteNodes(DeleteNodesRequest) returns (uprotocol.v1.UStatus) {
-    option (method_id) = 5;
-  }
+  /* Example: 
+  UnregisterUE(“//ultifi.gm.com/core.example/2.0")
+  Response: Status: code.OK
+ */
 
+ rpc UnregisterUE(UnregisterUERequest) returns ( uprotocol.v1.UStatus){
+  option (method_id) = 1;
+}
 
-  // Add one of more nodes to a parent node. If one of the nodes already exists, this API will return
-  // ALREADY_EXISTS and none of the nodes shall be added to the parent.
-  // **NOTE:** You MUST have write permission to the parent node
-  rpc AddNodes(AddNodesRequest) returns (uprotocol.v1.UStatus) {
-    option (method_id) = 6;
-  }
 
+  /* 
+  UEs shall call SetUETopics API to update discovery with the list of topics they service.
+  This will be used in two scenarios: 
+  1. when a UE is added on a device 
+  2. when UE wants to udpate the topic list
+  uDiscovery shall implement requisite permission model while processing this.
+ */
 
-  // Update a property in a node
-  // **NOTE:** You MUST have write permission to the node who's property you
-  // are updating
-  rpc UpdateProperty(UpdatePropertyRequest) returns (uprotocol.v1.UStatus) {
-    option (method_id) = 7;
-  }
+  /* Example: 
+  SetUETopics(parentUE:“///device/UE/1”, TopicList:["UE/1/topic1", "UE/1/topic2", "UE/1/rpc1"])
+  Response: Status: code.OK
 
+  SetUETopics(parentUE:“///device/UE2/1”, TopicList:["UE/1/topic1", "UE/1/topic2", "UE/1/rpc1"])
+  Response: Status: code.PERMISSION_DENIED (cant change other UEs)
 
-  // Register to receive Notifications to changes to one of more Nodes. The changes
-  // are published on the notification topic: '/core.udiscovery/3/nodes#Notification'
-  rpc RegisterForNotifications(NotificationsRequest) returns (uprotocol.v1.UStatus) {
-    option (method_id) = 8;
-  }
-  
+  */
 
-  // Unregister for Node change notifications
-  rpc UnregisterForNotifications(NotificationsRequest) returns (uprotocol.v1.UStatus) {
-    option (method_id) = 9;
+  rpc SetUETopics(SetUETopicsRequest) returns ( uprotocol.v1.UStatus){
+    option (method_id) = 1;
   }
 
-  
-  // Resolve a UUri filling in the missing names/numbers from the Discovery database.
-  // If the resolution was successful, the resolved UUri containing names and numbers
-  // is returned along with the following status.code values:
-  // - code.OK: Resolution was successfull
-  // - code.NOT_FOUND: Unable to find the missing names or numbers for the passed UUri
-  // - code.INVALID_ARGUMENT: The passed UUri is invalid (missing names or numbers)
-  // - code.PERMISSION_DENIED: The caller does not have permission to perform the resolution
-  rpc ResolveUri(ResolveUriRequest) returns (ResolveUriResponse) {
-    option (method_id) = 10;
-  }
-}
 
+  /* 
+  UEs shall call SetUEProperties API to update discovery with updated property values. 
+  For example, this could be used to update an UE's version.
```

Comment:
properties are part of the UUri, what other properties are we setting? Perhaps we need an explicit structure like UServiceTopic but more a UServiceProperties.